### PR TITLE
First version of structure outline

### PR DESCRIPTION
**Pull request by tmeasday**
I basically just whacked the existing structure and code-style outlines together, changing some details about JS structure to reflect current todos.

I'm not sure, is this the spot we should talk about using NPM with Meteor?
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/meteor/guide/pull/222%23issuecomment-179705907%22%2C%20%22https%3A//github.com/meteor/guide/pull/222%23issuecomment-180097906%22%2C%20%22https%3A//github.com/meteor/guide/pull/222%23issuecomment-180104067%22%2C%20%22https%3A//github.com/meteor/guide/pull/222%23discussion_r51956680%22%2C%20%22https%3A//github.com/meteor/guide/pull/222%23discussion_r51956850%22%2C%20%22https%3A//github.com/meteor/guide/pull/222%23discussion_r51962247%22%2C%20%22https%3A//github.com/meteor/guide/pull/222%23issuecomment-180278574%22%2C%20%22https%3A//github.com/meteor/guide/pull/222%23issuecomment-180278583%22%2C%20%22https%3A//github.com/meteor/guide/pull/222%23issuecomment-180480500%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/222%23issuecomment-179705907%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Currently%20we%20cover%20NPM%20in%20the%20build%20tool%20article.%20It%27s%20hard%20to%20say%20where%20people%20would%20look%20for%20that.%22%2C%20%22created_at%22%3A%20%222016-02-04T08%3A23%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ok%2C%20let%27s%20just%20make%20sure%20we%20link%20prominently%20to%20it%20from%20here%20then.%20Do%20we%20need%20to%20update%20that%20content%3F%22%2C%20%22created_at%22%3A%20%222016-02-04T23%3A15%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yes%2C%20it%20is%20currently%20written%20for%201.2%20with%20meteorhacks%3Anpm%20and%20cosmos%3Abrowserify%20and%20whatnot.%22%2C%20%22created_at%22%3A%20%222016-02-04T23%3A38%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ok%2C%20I%27ve%20added%20a%20todo%20to%20revise%20that%20content.%22%2C%20%22created_at%22%3A%20%222016-02-05T09%3A57%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222016-02-05T09%3A57%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%2C%20%7B%22body%22%3A%20%22Looks%20awesome%21%20Still%20curious%20about%20the%20absolute%20vs.%20relative%20path%20issue%2C%20but%20good%20to%20go%20otherwise%21%22%2C%20%22created_at%22%3A%20%222016-02-05T18%3A20%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20f7cd21c3823b3f7d39553646dcbcf00fdddfcd80%20outlines/structure.md%208%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/222%23discussion_r51956680%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Do%20we%20recommend%20importing%20files%20via%20absolute%20or%20relative%20paths%3F%22%2C%20%22created_at%22%3A%20%222016-02-04T23%3A40%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20outlines/structure.md%3AL1-62%22%7D%2C%20%22Pull%20f7cd21c3823b3f7d39553646dcbcf00fdddfcd80%20outlines/structure.md%2012%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/222%23discussion_r51956850%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22We%20should%20mention%20tests%20as%20well%2C%20to%20emphasize%20that%201.%20you%20should%20have%20tests%20and%202.%20you%20should%20put%20them%20next%20to%20the%20thing%20they%20test%22%2C%20%22created_at%22%3A%20%222016-02-04T23%3A41%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222016-02-05T00%3A47%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20outlines/structure.md%3AL1-62%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/222#issuecomment-179705907'>General Comment</a></b>
- <a href='https://github.com/stubailo'></a> Currently we cover NPM in the build tool article. It's hard to say where people would look for that.
- <a href='https://github.com/tmeasday'></a> Ok, let's just make sure we link prominently to it from here then. Do we need to update that content?
- <a href='https://github.com/stubailo'></a> Yes, it is currently written for 1.2 with meteorhacks:npm and cosmos:browserify and whatnot.
- <a href='https://github.com/tmeasday'></a> Ok, I've added a todo to revise that content.
- <a href='https://github.com/tmeasday'></a> :+1:
- <a href='https://github.com/stubailo'></a> Looks awesome! Still curious about the absolute vs. relative path issue, but good to go otherwise!
- [ ] <a href='#crh-comment-Pull f7cd21c3823b3f7d39553646dcbcf00fdddfcd80 outlines/structure.md 8'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/222#discussion_r51956680'>File: outlines/structure.md:L1-62</a></b>
- <a href='https://github.com/stubailo'></a> Do we recommend importing files via absolute or relative paths?
- [x] <a href='#crh-comment-Pull f7cd21c3823b3f7d39553646dcbcf00fdddfcd80 outlines/structure.md 12'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/222#discussion_r51956850'>File: outlines/structure.md:L1-62</a></b>
- <a href='https://github.com/stubailo'></a> We should mention tests as well, to emphasize that 1. you should have tests and 2. you should put them next to the thing they test

<a href='https://www.codereviewhub.com/meteor/guide/pull/222?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/meteor/guide/pull/222?approve=1'></a>&nbsp;<a href='https://github.com/meteor/guide/pull/222'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by stubailo** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/meteor/guide/pull/222'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by stubailo** [comment]
Currently we cover NPM in the build tool article. It's hard to say where people would look for that.

**Comment by tmeasday** [comment]
Ok, let's just make sure we link prominently to it from here then. Do we need to update that content?

**Comment by stubailo** [comment]
Yes, it is currently written for 1.2 with meteorhacks:npm and cosmos:browserify and whatnot.

File: outlines/structure.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Application structure and code style
+
+1. What's different between Meteor and other kinds of apps?
+    1. All JavaScript, enables code sharing
+    2. Different envrironment: Client vs. server vs. both
+    3. Module system is new, but we recommend its use (answer to why `imports/`?)
+2. General JavaScript structure
+    1. All code in `imports/` apart from initializers in `main/`
```
Comment (stubailo) [inline outlines/structure.md]: Do we recommend importing files via absolute or relative paths?

File: outlines/structure.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    2. Group code into data - `api/` and rendering - `client` (?)
+    1. Directory structure around features, not client/server
+    2. LESS/SCSS files are in the same directory as components
+    3. One file per unit - template, method, collection, etc
```
Comment (stubailo) [inline outlines/structure.md]: We should mention tests as well, to emphasize that 1. you should have tests and 2. you should put them next to the thing they test

Comment (tmeasday) [inline outlines/structure.md]: :+1:

**Comment by tmeasday** [comment]
Ok, I've added a todo to revise that content.

**Comment by tmeasday** [comment]
:+1:

**Comment by stubailo** [comment]
Looks awesome! Still curious about the absolute vs. relative path issue, but good to go otherwise!

**Comment by tmeasday** [comment]
Ok, merged. Will continue to try and find that out :)

**Pull request merged**